build-system/release-tagger/create-release.js

```diff
@@ -23,20 +23,113 @@
  */
 
 const argv = require('minimist')(process.argv.slice(2));
+const dedent = require('dedent');
 const {
   createRelease,
   getPullRequestsBetweenCommits,
   getRelease,
 } = require('./utils');
 const {GraphQlQueryResponseData} = require('@octokit/graphql'); //eslint-disable-line no-unused-vars
 
+/**
+ * Format pull request line
+ * @param {GraphQlQueryResponseData} pr
+ * @return {string}
+ */
+function _formatPullRequestLine(pr) {
+  const {mergeCommit, number, title, url} = pr;
+  const {abbreviatedOid, commitUrl} = mergeCommit;
+  return dedent`\
+  <a href="${commitUrl}"><code>${abbreviatedOid}</code></a> ${title} \
+  (<a href="${url}">#${number}</a>)`;
+}
+
+/**
+ * Create raw notes
+ * @param {Array<GraphQlQueryResponseData>} prs
+ * @return {Array<string>}
+ */
+function _createRawNotes(prs) {
+  return prs.map((pr) => _formatPullRequestLine(pr));
+}
+
+/**
+ * Organize pull requests into sections
+ * @param {Array<GraphQlQueryResponseData>} prs
+ * @return {Array<string>}
+ */
+function _createSections(prs) {
+  const sections = {
+    'ads': [],
+    'build-system': [],
+    'src': [],
+    'third_party': [],
+    'validator': [],
+    'package updates': new Array(), // [] defaults to never[], which tsc flags
```

Comment:
Can you define a type for this at the beginning of the file, and then not have to "hack" a `new Array` where every other field is a `[]`? 😅
```javascript
/** @typedef {{
 *    ads: !Array<string>,
 *    "build-system": !Array<string>,
 *    src: !Array<string>,
 *    third_party: !Array<string>,
 *    "package updates": !Array<string>,
 *  }}
 */
let SectionsDef;
```

or
```javascript
/** @typedef {{
 *    [key: string]: !Array<string>,
 *  }}
 */
let SectionsDef;
```

Reply:
I tried adding a typedef, it still didn't work. It has a problem with `package updates` being the key. `package-updates` works fine.